### PR TITLE
#351: Schema validator updates file with invalid output when behind VPN

### LINKED ISSUE #351: Schema validator updates file with invalid output when behind VPN

### LINKED ISSUE #362: Logger not logging in console with python 3.9
Something appears to have changed in the behavior of the `logger` class between 3.8 and 3.9

### PR DESCRIPTION
**Pull request by tlamonthezie**
Fixes #351
Fixes #362 (Logger level now set by default to info if not given. seems to resolve also this logger issue with Python3.9)

This PR includes
- Content-Type checking for the SchemaValidator download. If bad content-type (like text/html) the application will use the latests downloaded SchemaValidator if exists and will show an error in the output but application will run.
- Some refactoring of code in LBAF_App.py by encapsulating configuration loading logic in a Loader class (more object oriented)

@lifflander I let you verify if it works better on your side. To test on your side with your proxy : 
1. Check you already have a valid  `src/lbaf/imported/JSON_data_files_validator.py` file (like in vt repo).
2. Activate your proxy (responsible on the html message when LBAF tries to download the SchemaValidator script)
3. Run LBAF. You should get the error in the output but from now LBAF should continue to run by using the existing script

### CODE REVIEW

File: src/lbaf/Applications/LBAF_app.py
Comment (ppebay) [inline src/lbaf/Applications/LBAF_app.py:1]: Bad option value `W0718` revealed by GitHub action; other Github actions findings (e.g. bad placement of `import` statement).

File: src/lbaf/Utils/logger.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def msgcolor(msg):
 
 def logger(
         name: str = "root",
-        level: Union[str, None] = None,
+        level: Union[str, None] = 'info',
```
Comment (ppebay) [inline src/lbaf/Utils/logger.py:69]: Please use double quotes for strings that are longer than a single character (this is one of the unwritten style rules of LBAF)

File: src/lbaf/Applications/LBAF_app.py [hunk 2]
```diff
@@ -21,77 +21,139 @@
...
-        except HTTPError as err:
+class Loader:
+    """Class to load LBAF configuration file and SchemaValidator script"""
+    def __resolve_config_file(self) -> str:
+        """Parses command line argument and resove the config file path."""
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--config",
+            help="Path to the config file. If path is relative it must be resolvable from either the current working "
+                "directory or the config directory",
+            default='conf.yaml'
```
Comment (ppebay) [inline src/lbaf/Applications/LBAF_app.py:36]: use double quotes

File: src/lbaf/Applications/LBAF_app.py [hunk 3]
```diff
@@ -21,77 +21,139 @@
...
-            script_url = f"https://raw.githubusercontent.com/DARMA-tasking/vt/develop/scripts/{script_name}"
-            filename = urlretrieve(script_url, os.path.join(import_dir, script_name))
-            print(f"{module_name} Saved SchemaValidator to: {filename}")
-        except HTTPError as err:
+class Loader:
+    """Class to load LBAF configuration file and SchemaValidator script"""
+    def __resolve_config_file(self) -> str:
+        """Parses command line argument and resove the config file path."""
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--config",
```
Comment (ppebay) [inline src/lbaf/Applications/LBAF_app.py:33]: I suggest that we use a non-abbreviated option (`--configuration`) for the long form of the this flag, possibly with an added short form as well for convenience (`-c`).

File: src/lbaf/Applications/LBAF_app.py [hunk 4]
```diff
@@ -21,77 +21,139 @@
...
+        else:
             sys.excepthook = exc_handler
-            raise ConnectionError(f"Can not download file: {err.filename} \n"
-                                  f"Server responded with code: {err.fp.code} and message: {err.fp.msg}") from err
-        except URLError as err:
+            raise FileNotFoundError("Please provide path to the config file with '--config' argument.")
+
+        if not os.path.isfile(config_file):
             sys.excepthook = exc_handler
-            raise ConnectionError('Probably there is no internet connection') from err
```
Comment (ppebay) [inline src/lbaf/Applications/LBAF_app.py:72]: ' --> "

File: src/lbaf/Applications/LBAF_app.py [hunk 5]
```diff
@@ -21,77 +21,139 @@
...
+                content_type = http_message.get_content_type()
+                # validate content type for script that has been retrieved
+                if content_type == 'text/plain':
+                    os.rename(tmp_filename, filename)
+                    logger().info("Saved SchemaValidator to: %s", filename)
+                else:
+                    os.remove(tmp_filename)
+                    if os.path.isfile(filename):
+                        logger().error(
+                            'Unexpected Content-Type (%s) for SchemaValidator file.'
```
Comment (ppebay) [inline src/lbaf/Applications/LBAF_app.py:118]: Please use f-strings and double quotes

**Review by ppebay — CHANGES_REQUESTED** [review]
* more comments needed to explain code
* f-strings should be used rather than `%s`
* style rule: use double-quotes for non-single-character strings

File: src/lbaf/Applications/LBAF_app.py [hunk 6]
```diff
@@ -21,77 +21,139 @@
...
+    def check_and_get_schema_validator(self, overwrite_validator: bool = True):
+        """Makes sure that SchemaValidator can be imported, and it's the latest version available."""
+        def save_schema_validator_and_init_file(import_dir: str):
+            """Downloads the latest version of the Schema Validator to the import directory"""
+            # create empty __init__.py file
+            with open(os.path.join(import_dir, "__init__.py"), 'wt', encoding='utf-8'):
+                pass
+            # then download the SchemaValidator for vt files
+            try:
+                script_name = "JSON_data_files_validator.py"
```
[the remark this answers is not in the document]
Comment (tlamonthezie) [reply src/lbaf/Applications/LBAF_app.py]: Hello Philippe, I finally moved the schema validator download logic to a separated module I think it is more clear

**Review by ppebay — APPROVED** [review]
Second review approved

**Pull request merged**